crates/orchestrator/src/jobs/snos_job/mod.rs

```diff
@@ -29,28 +51,133 @@ impl Job for SnosJob {
+                strk_l1_data_gas_price: NonZeroU128::new(0).unwrap(),
+            },
+            use_kzg_da: snos_input.general_config.use_kzg_da,
+        };
+
+        let chain_info = ChainInfo {
+            chain_id: snos_input.general_config.starknet_os_config.chain_id.clone(),
+            fee_token_addresses: FeeTokenAddresses {
+                eth_fee_token_address: snos_input.general_config.starknet_os_config.fee_token_address,
+                // TODO: assert that the STRK fee token address is deprecated_fee_token_address
+                strk_fee_token_address: snos_input.general_config.starknet_os_config.deprecated_fee_token_address,
+            },
+        };
+
+        let block_context = match pre_process_block(
+            &mut state,
+            Some(block_number_and_hash),
+            block_info,
+            chain_info,
+            VersionedConstants::latest_constants().clone(),
+        ) {
+            Ok(block_context) => block_context,
+            Err(e) => return Err(eyre!("pre_process_block failed for block #{}: {}", block_number, e)),
+        };
+
+        // TODO: contract_storage_map should be retrieved from where?
+        let contract_storage_map = HashMap::default();
+        let execution_helper = ExecutionHelperWrapper::new(
+            contract_storage_map,
+            vec![], // TODO: vec of TransactionExecutionInfo, how to get it?
+            &block_context,
+            (Felt252::from_u64(block_number.0).unwrap(), snos_input.block_hash),
+        );
+
+        // 3. Import SNOS in Rust and execute it with the input data
+        let (_cairo_pie, _snos_output) = match run_os(
+            // TODO: what is this path?
+            String::from("PATH/TO/THE/OS"),
+            // TODO: which layout should we choose?
+            LayoutName::plain,
+            snos_input,
+            block_context,
+            execution_helper,
+        ) {
+            Ok((cairo_pie, snos_output)) => (cairo_pie, snos_output),
+            Err(e) => return Err(eyre!("Could not run SNOS for block #{}: {}", block_number, e)),
+        };
+
         // 3. Store the received PIE in DB
+        // TODO: Store the PIE & the SnosOutput once S3 is implemented
         todo!()
     }
 
     async fn verify_job(&self, _config: &Config, _job: &mut JobItem) -> Result<JobVerificationStatus> {
         // No need for verification as of now. If we later on decide to outsource SNOS run
         // to another servicehow a, verify_job can be used to poll on the status of the job
-        todo!()
+        Ok(JobVerificationStatus::Verified)
     }
 
     fn max_process_attempts(&self) -> u64 {
-        todo!()
+        1
     }
 
     fn max_verification_attempts(&self) -> u64 {
-        todo!()
+        // TODO: isn't 10 a lot?
+        10
     }
 
     fn verification_polling_delay_seconds(&self) -> u64 {
-        todo!()
+        // TODO: what is an average run time for SNOS?
+        60
```

Comment:
we don't know the runtime yet but in the current implementation looks like we run in synchronously so verification delay can be like 1s

Reply:
Ah yes true - updated